coin-change/std-freejia.java

```diff
@@ -0,0 +1,54 @@
+/** 
+ * BFS 는 너비 우선이므로, 가장 얕은 깊이로 목표 금액을 만족하면 '최소 동전 개수'를 사용한 것입니다. 
+ * 너비 == 사용한 동전 개수 
+ * 
+ * Queue<Integer> 동전 금액 '합' 을 관리합니다 
+ * 
+ * 목표 금액이 amount 일 때, 
+ * 0 부터 amount 까지의 합을 체크할 boolean[] 을 활용합니다. 
+ * 합계 금액이 '이미 만들어본 합' 이면 boolean[] 에 체크하여 중복방문을 방지하기 위함.
+ * 
+  */
+class Solution {
+    public int coinChange(int[] coins, int amount) {
+        if (amount == 0) return 0; // 목표 금액 0인 경우, '0개' 리턴
+
+        Arrays.sort(coins); // 작은 동전부터 더해본다
+
+        // 이미 만들어본 '합' 은 지나가도록.
+        boolean[] visitedAmount = new boolean[amount + 1];
```

Comment:
반복문을 통한 순차 탐색 대신 BFS를 통해 평균적인 시간을 단축하신 것 같습니다.
해시 자료구조를 사용하면 평균적인 공간도 줄일 수 있을 것 같군요.

```suggestion
        Set<Integer> visitedAmount = new HashSet<>();
```